### DISCUSSION TITLE
Featurecounts

### DISCUSSION THREAD

**Discussion by gsk7757** [biostars/biostar-handbook-issues#311] [Q&A]
Hello, 
I have a question about the output of FeatureCounts. Does the output .txt file from the Featurecounts calculated based on TPM or do I have to calculate the TPM from the counts.txt file?

**Comment by ialbert** [comment]
featureCounts computes raw readcounts

in general with all tools you should consult the manual, that's the ultimate resource in all of these cases, anyone stating something might be wrong, after all, there are so many tools and techniques out there

> featureCounts takes as input SAM/BAM files and an annotation file including chromosomal coordinates of features. It outputs numbers of reads assigned to features (or meta-features). It also outputs stat info for the overall summrization results, including number of successfully assigned reads and number of reads that failed to be assigned due to various reasons (these reasons are included in the stat info)

https://subread.sourceforge.net/featureCounts.html

**Reply by gsk7757** [reply]
Thank you. Just a clarification, can the raw read counts generated by the featurecounts be used for comparative analysis or do they have to be converted to TPM/RPKM for the analysis? Also, can the input into the deseq2/edgeR be just raw read counts or do they have to be converted?

**Reply by ialbert** [reply]
in general never use TPM, RPKM ever, those are invalid measures - it is a awkward turn of science that they even exist

it shows how even scientists can misunderstand the very topics they study, first they proposed RPKM, then they "fixed"it with TMP but both are wrong measures - technically these measures only existed for about five years until people fully realized that none work - but by then got entrenched and are still in use even though are not valid ways to do anything

Always use raw counts and let the statistical methods operate on raw counts

**Reply by gsk7757** [reply]
Excellent. Thank you so much for your time and help.
